src/modules/scraper/service.ts

```diff
@@ -30,6 +33,9 @@ export class ScraperService {
         if (this.appConfig.values.enableSpokePoolsEventsProcessing) {
           await this.publishBlocks();
         }
+        if (this.appConfig.values.enableMerkleDistributorEventsProcessing) {
+          await this.publishMerkleDistributorBlocks();
```

Comment:
This is just an observation, but it's weird that `this.publishBlocks()` and `this.publishMerkleDistributorBlocks()` looks different. Basically they do the same thing (only the repositories and the configured start block are different), but one is a 10 line function and the other one looks more complicated than it actually is. I don't have a problem leaving the code as it is, but I see an opportunity here in organising the code a little bit better. We can chat in private about it. It's clear that the original code wasn't as flexible and modular as it had to be

Reply:
I refactored it a bit to reuse the same functions. https://github.com/across-protocol/scraper-api/commit/0cb04e35fd02e4099c1bd24245b8fbbecb6a129d